src/hotspot/share/opto/type.cpp

```diff
@@ -3374,93 +3381,93 @@ void TypePtr::InterfaceSet::verify() const {
 }
 #endif
 
-TypePtr::InterfaceSet TypeOopPtr::InterfaceSet::union_with(const InterfaceSet& other) const {
-  InterfaceSet result;
+const TypeInterfaces* TypeInterfaces::union_with(const TypeInterfaces* other) const {
+  GrowableArray<ciInstanceKlass*> result_list;
   int i = 0;
   int j = 0;
-  while (i < _list.length() || j < other._list.length()) {
+  while (i < _list.length() || j < other->_list.length()) {
     while (i < _list.length() &&
-           (j >= other._list.length() ||
-            compare(_list.at(i), other._list.at(j)) < 0)) {
-      result.raw_add(_list.at(i));
+           (j >= other->_list.length() ||
+            compare(_list.at(i), other->_list.at(j)) < 0)) {
+      result_list.push((ciInstanceKlass*)_list.at(i));
```

Comment:
I think you should use `as_instance_klass()` here and in other code.

Reply:
I moved both `_list` and `result_list` to `ciInstanceKlass` so there's no cast anymore.